strawberry/experimental/pydantic/utils.py

```diff
@@ -118,3 +120,18 @@ def get_default_factory_for_field(field: ModelField) -> Union[NoArgAnyCallable,
         return lambda: None
 
     return UNSET
+
+
+def ensure_all_auto_fields_in_pydantic(
+    model: Type[BaseModel], auto_fields: Set[str], cls_name: str
+) -> Union[NoReturn, None]:
+    # Raise error if user defined a strawberry.auto field not present in the model
+    undefined = [
+        field_name for field_name in auto_fields if field_name not in model.__fields__
+    ]
+    if len(undefined) > 0:
```

Comment:
This might be even better!

```suggestion
    non_existing_fields = auto_fields - model.__fields__.keys()
    
    if len(non_existing_fields) > 0:
```